### ISSUE TITLE
EmbeddedFileProvider priority

### ISSUE THREAD

**Issue by rjamesnw** [aspnet/Mvc#6108]
Currently the `EmbeddedFileProvider` works great to pull a file from a referenced assembly, but I want the _local_ .cshtml files in the same path for the main project to override the defaults in the assembly (should be found first). I tried adding a `PhysicalFileProvider` before the embedded one, then I tried `HostingEnvironment.ContentRootFileProvider`, but in both cases then nothing worked at all. Is there a way to have the physical files found first before assemblies?  I'm still looking through the code, but it's designed in a way  that is very hard to follow what is going on (with all the caching and callbacks, etc.).

```
// options.FileProviders.Add(HostingEnvironment.ContentRootFileProvider); // (tried with and without this)
options.FileProviders.Add(new EmbeddedFileProvider(typeof(TypeInAssembly).GetTypeInfo().Assembly));
```

**Comment by rjamesnw** [comment]
Sorry, I just realized "nothing worked at all" was no very specific. ;)  Anyhow, I figured it out; turns out the embedded provider worked to pulled the "_ViewImports.cshtml" from the assembly; however, my "_ViewImports.cshtml" in my MAIN project had an error, but the error was about a file not found, and I assumed it was the file providers, but it was not. ;)  Anyhow, all is well.

**Comment by rynowak** [comment]
@rjamesnw so is there an issue here or not?

**Comment by rjamesnw** [comment]
Sorry, there's no issue now, I figured it out, go ahead and close this.
